### ISSUE TITLE
Reactive Transactions: Example in guide is not working

### ISSUE THREAD

**Issue by Sgitario** [quarkusio/quarkus#16433]
## Describe the bug
I tried to replicate the same example than in https://quarkus.io/guides/reactive-sql-clients#transactions:

```java
public static Uni<Void> insertTwoFruits(PgPool client, Fruit fruit1, Fruit fruit2) {
    return SqlClientHelper.inTransactionUni(client, tx -> {
        Uni<RowSet<Row>> insertOne = tx.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING id")
                .execute(Tuple.of(fruit1.name));
        Uni<RowSet<Row>> insertTwo = tx.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING id")
                .execute(Tuple.of(fruit2.name));

        return insertOne.and(insertTwo)
                // Ignore the results (the two ids)
                .onItem().ignore().andContinueWithNull();
    });
}
```

When calling this method, my app fails with:

```
ERROR: Mutiny had to drop the following exception
io.vertx.core.VertxException: Transaction already completed
	at io.vertx.sqlclient.impl.TransactionImpl.checkPending(TransactionImpl.java:115)
	at io.vertx.sqlclient.impl.TransactionImpl.rollback(TransactionImpl.java:154)
	at io.vertx.sqlclient.impl.TransactionImpl.rollback(TransactionImpl.java:159)
	at io.vertx.mutiny.sqlclient.Transaction.lambda$rollback$2(Transaction.java:112)
	at io.smallrye.mutiny.vertx.AsyncResultUni.subscribe(AsyncResultUni.java:30)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:37)
	at io.smallrye.mutiny.operators.uni.UniOnItemTransformToUni.subscribe(UniOnItemTransformToUni.java:25)
	at io.smallrye.mutiny.operators.AbstractUni.subscribe(AbstractUni.java:37)
	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap$UniOnItemOrFailureFlatMapProcessor.performInnerSubscription(UniOnItemOrFailureFlatMap.java:99)
	at io.smallrye.mutiny.operators.uni.UniOnItemOrFailureFlatMap$UniOnItemOrFailureFlatMapProcessor.onFailure(UniOnItemOrFailureFlatMap.java:65)
	at io.smallrye.mutiny.operators.uni.UniOperatorProcessor.onFailure(UniOperatorProcessor.java:44)
	at io.smallrye.mutiny.operators.uni.UniAndCombination$AndSupervisor.check(UniAndCombination.java:97)
	at io.smallrye.mutiny.operators.uni.UniAndCombination$UniHandler.onFailure(UniAndCombination.java:180)
	at io.smallrye.mutiny.vertx.AsyncResultUni.lambda$subscribe$1(AsyncResultUni.java:36)
	at io.vertx.mutiny.sqlclient.PreparedQuery$2.handle(PreparedQuery.java:140)
	at io.vertx.mutiny.sqlclient.PreparedQuery$2.handle(PreparedQuery.java:135)
```

## To Reproduce

1.  git clone `https://github.com/Sgitario/quarkus-examples`
2.  cd quarkus-examples/reproducers/reactive-sql
3. mvn clean verify -Dtest=FruitResourceTest#shouldInsertTwoFruits

The test should pass, but it's failing with the above exception

### Quarkus version or git rev
999-SNAPSHOT or 1.13.1.Final or later

**Comment by geoand** [comment]
cc @tsegismont

**Comment by tsegismont** [comment]
This looks like a duplicate of https://github.com/quarkusio/quarkus/issues/16430

There is an issue upstream that seems related https://github.com/eclipse-vertx/vertx-sql-client/issues/930

**Comment by vietj** [comment]
do you know hwat is failing the transaction ? I don't get why inserting two valid fruits is making this fail

**Comment by cescoffier** [comment]
Closing as it's fixed in #16430. Please re-open if it still happening.
